core/trino-main/src/main/java/io/trino/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -267,6 +268,13 @@ public Expression visitWindow(WindowNode node, Void context)
             return node.getSource().accept(this, context);
         }
 
+        @Override
+        public Expression visitPatternRecognition(PatternRecognitionNode node, Void context)
+        {
+            Expression sourcePredicate = node.getSource().accept(this, context);
+            return pullExpressionThroughSymbols(sourcePredicate, node.getOutputSymbols());
```

Comment:
There's probably some room for improvement in the future by taking advantage of the label predicates.

Reply:
Label predicates capture dependencies between different rows, while the effective predicate applies to every row independently. It is possible though that some row-level properties could be deduced from the label conditions.
When doing that, we must be cautious about empty matches and unmatched rows. They produce output rows that don't have a label.

There is further room for improvement, too:
- extract predicate for `classifier` - e.g. `classifier IN ('A', 'B', 'C')`
- extract predicate for `match_number` - this is a number ranging from 1 and limited by the partition size
- extract predicate for other measure columns.